### DISCUSSION TITLE
Restoring deleted files based off initial commit?

### DISCUSSION THREAD

**Discussion by ejrobinson** [community/community#24086] [Apps, API and Webhooks]
<p>Hi,</p>
<p>Made a stupid mistake today and deleted a local repository before it had pushed to the web. Figured deleting it locally wouldn’t actually remove all the local files without sending them to some kind of recycle bin.</p>
<p>I’ve managed to restore the local repository and so have essentially the contents of all the files in the initial commit notes but none of these files exist on disk any more</p>
<p>Is there any way, bar going through and copy-pasting the contents of each file one file at a time to restore the files to disk?</p>
<p>Many thanks</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="ejrobinson" data-post="1" data-topic="162472">
<div class="title">
<div class="quote-controls"></div>
 ejrobinson:</div>
<blockquote>
<p>I’ve managed to restore the local repository and so have essentially the contents of all the files in the initial commit notes but none of these files exist on disk any more</p>
</blockquote>
</aside>
<p>In what form do you have the local repository now? If you have the <code>.git</code> folder that should be enough to check out the files again.</p>

**Comment by ejrobinson** [comment]
<p>I have the .git folder so I have re-added the repository to the desktop client and can see the history with the initial commit.</p>
<p>The files which would have resided in the parent folder for the .git no longer exist. I have tried pulling the initial branch through the client, but it requires the files to be committed first which doesn’t work</p>

**Comment by ejrobinson** [comment, marked as answer]
<p>This was enough to nudge me in the right direction.</p>
<p>The GitHub desktop client couldn’t do what I wanted but by opening the GIT GUI in the folder I could create a new branch from the initial commit and then check out that new branch restoring the files from the .git/objects</p>
<p>Cheers</p>
